request.go

```diff
@@ -465,3 +489,15 @@ func (r *Request) fmtBodyString() (body string) {
 
 	return
 }
+
+func (r *Request) selectAddr(addrs []*net.SRV, path string, attempt int) string {
+	if addrs == nil {
```

Comment:
Can you please add test case for when `addrs` is `nil` and SRV is record is set. So we can ensure request goes through or gives error info to caller.